### ISSUE TITLE
`translate: false` in a child of a structure field is saved as `null` on secondary language

### ISSUE THREAD

**Issue by yandodov** [getkirby/kirby#1602]
**Describe the bug**
I was eager to test this feature of Kirby 3.1.0:
>translate: false is no longer ignored in structure fields

But it seems there's a problem with it. When you modify a structure entry that has a `translate: false` field in a secondary language, that field's value is not saved.

**To Reproduce**
1. Create a page with the following template:
```yaml
title: Test Page
preset: page
fields:
  title:
    type: text
  mydata:
    type: structure
    fields:
      foo:
        type: text
      bar:
        type: text
        translate: false
```
2. In the panel, create a new page using the new template.
3. In the default language, add a new entry in the structure.
4. Save your changes.
5. Switch to a secondary language.
6. Open the newly added structure field entry. The `translate: false` field should correctly be disabled and have the default language value, while the other field is editable.
7. Change the editable field.
8. Save your changes.
9. The `translate: false` field of the entry is suddenly empty and if you open the secondary language's txt file, you can see that the field's value is `null`. The other field (the translatable one) is alright, though. However, since the field has `translate: false` and is disabled, you can no longer give it a value via the panel.

**Expected behavior**
When you click save in step 8, it should either:
1. Copy the default language's untranslatable fields instead of setting `null` for their values.
2. Save only the translatable fields and merge the default language's data with the translated data.

**Kirby Version**
3.1.0

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version 73.0.3683.75 (Official Build) (64-bit)

**Comment by yandodov** [comment]
Regarding the expected behavior options I mentioned above.

At first glance, I think that untranslatable fields should not be saved in the secondary languages' txt files at all. That's how other fields work - if `myImage` has `translate: false`, it's value is not saved in `de.txt`. When Kirby prepares the page for render, it automatically pulls the value from `en.txt` (the default language txt file).

For structure fields, I think it should be the same. When you save a translated structure field, it should save only the translatable fields. Later when the page is rendered, Kirby should pull the structures from both files, parse their YAML contents, and merge the resulting arrays, with the translated fields being merged on top of the default language ones.

This is better because if you change an untranslatable field in a structure under the default language, the translated pages would be affected and synced because they use the default values.

I see can only one issue with that and that's when you add a new entry in the default language and change its order to be first. The translation would only have the translatable fields for entry 1 at index 0, while the default would have entry 2 at index 0 and entry 1 at index 1. So when you open the translated page, the translations of entry 1 would be merged with the default values of entry 2. That could be fixed by syncing the translations each time the defaults are updated? Might be tricky, but certainly doable.

**Edit:** If a structure field has `translate:false`, its translations should be updated accordingly so that they can be merged successfully. For example, if you have 5 entries and you move entry at index 3 to index 0, Kirby should modify all translations and move the data at index 3 to index 0 too. This way, when the default data is merged with the translation, the right entries will be merged.

**Comment by distantnative** [comment]
I don't think we can solve this.

The matching of translations with the structure field is still on a per-field basis.
There is no matching between entries.

Not:

| Language 1 | Language 2 |
|-|-|
| A1 | A2| 
| B1 | B2| 

But: 

| Language 1 | Language 2 |
|-|-|
| A | C | 
| B | D | 

The structure fields/Kirby has no concept of entry C being the direct translation of entry A.

I think we might have to outrule the `translate` option within structure fields in general.

Thoughts, @bastianallgeier, @lukasbestle, @texnixe?

**Comment by yandodov** [comment]
@distantnative you're right. That's why in my [kirby-outsource](https://github.com/OblikStudio/kirby-outsource/blob/dev/src/Marker.php) plugin ([docs](https://github.com/OblikStudio/kirby-outsource#structure-synchronization)), I have a hook that adds IDs to structure entries on each save. When you add an entry in language A, it receives an ID and is added to the other languages. Then, that entry has the same ID in all translations and is uniquely identified. That way, you can keep track. When you mutate an entry in one language, you know which entry received the mutation in the other translations.

This, of course, enforces that every translation should have the same number of entries, which might not be desired always. For that reason, I propose to have a `sync: true/false` option that toggles this behavior. Then, if you have `translate: false` on a structure field without `sync: true`, you get an error.

---

I haven't yet made much noise about kirby-outsource, but I use it as a dependency in [kirby-memsource](https://github.com/OblikStudio/kirby-memsource) and you can install that plugin to see it in action. For now, it works great. It even handles nested structures nicely.

**Comment by lukasbestle** [comment]
Such a feature to enforce item synchronization between translations via a unique item ID would be great. But that's more a new feature (= idea issue) than a bug.

I think we are on the same page that fixing `translate: false` with the current implementation is impossible. So maybe we should disable `translate: false` again for structure fields until we have the item synchronization feature. I wonder why we added incomplete support for it in v3.1.0 in the first place. 🤔

**Comment by texnixe** [comment]
I'm not so sure disabling it again is a good idea, because I think that there are definitely use cases where the complete field should be disabled in secondary languages, e.g. a list of addresses or registrations. Or do you mean on the subfield level, @lukasbestle?

**Comment by lukasbestle** [comment]
Yes, I mean the subfield level. The current behavior that allows single fields of a structure field to be disabled doesn't make sense.

**Comment by lukasbestle** [comment]
@distantnative @bastianallgeier What do you think about disabling `translate: false` for subfields of structure fields until there is structure item synchronization (see https://github.com/getkirby/ideas/issues/453)?

**Comment by alancwoo** [comment]
I just encountered this issue now as well in `3.3.6`. We have a fairly complex structure field with something like:

`Image`, `Marker X Position`, `Marker Y Position`, `Marker Size`, `Caption`

Basically, only the `Caption` needs to be translated, so I set all the other fields to `Translate: false`, and was surprised to see the full row go to null once the caption translations were entered. 

It's odd that `translate` is mentioned in the documents if it's not quite functional as mentioned - especially actually as allowing translation prevents the entire row from being nulled.
